### PR TITLE
feat(label): add property

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/core/lv_obj_property.h [hunk 1]
```diff
@@ -56,15 +56,16 @@ enum {
     LV_PROPERTY_ID_INVALID      = 0,
 
     /*ID 0x01 to 0xff are style ID, check lv_style_prop_t*/
```
Comment (W-Mai) [inline src/core/lv_obj_property.h]:
```suggestion
    /*ID 0x00 to 0xff are style ID, check lv_style_prop_t*/
```

**Review by W-Mai — CHANGES_REQUESTED** [review]

File: src/core/lv_obj_property.h (hunk 1 above)
Comment (XuNeo) [reply src/core/lv_obj_property.h]: It's actually `0x01`. See
```c
enum _lv_style_prop_t {
    LV_STYLE_PROP_INV               = 0,

    /*Group 0*/
    LV_STYLE_WIDTH                  = 1,
```

We use `LV_PROPERTY_STYLE_START` 0 as enum start. It's a bit counter intuitive.

**Review by kisvegabor — APPROVED** [review]
Great, thanks!

I wonder if we can generate these automatically somehow. I thin for most of the properties it should work, but I don't know how to handle 
- objects created by `lv_*_add_*` 
  - chart series
  - scale section
  - messagebox title, footer, etc (probably ok, as these are `lv_obj`s)
  - tabview tabs (`lv_obj`s too)
  - window buttons and titles (`lv_obj`s too)
- more complex setters 
  - buttonmatrix maps 
  - line points
  - table cell data
  - spans

**Review by liamHowatt — APPROVED** [review]

**Comment by XuNeo** [comment]
> I wonder if we can generate these automatically somehow. I thin for most of the properties it should work, but I don't know how to handle

The property can only work with simply value. If it's too complex, like an array of points, pointers, it should be handled case by case.

> more complex setters

The original purpose of property API is to unify and simplify binding language. If it's not unified, then it's not achievable through property API. 

E.g.
```c
void lv_roller_set_options(lv_obj_t * obj, const char * options, lv_roller_mode_t mode);
void lv_roller_set_selected(lv_obj_t * obj, uint32_t sel_opt, lv_anim_enable_t anim);

```

**Comment by kisvegabor** [comment]
For the editor we should have a parseable and simple API for all the widgets which maps to properties in a declarative language. So ideally each 
- create function should have a single argument: the parent
- setter should have a two argument: the target and the value

Still considering if it's possible in all cases.

**Comment by XuNeo** [comment]
For luavgl, I provide a method to override LVGL's default property API, extend those special cases.

Given a provided property name

1. Try luavgl's widget property table
2. Try lvgl's property table
3. Try lvgl's obj style.

https://github.com/XuNeo/luavgl/pull/50/commits/2c3fc0ef32e46ff6a577c5170058e615a6f3db4c#diff-7e3a522db40999cfa49a42764aef9413eeac47ecaf8e21d5f02582c09820b9dbR716-R750

**Pull request merged**